python/llm/src/bigdl/llm/ggml/model/generation/utils.py

```diff
@@ -30,33 +31,64 @@ class GenerationMixin:
 
     Pass custom parameter values to 'generate' .
     """
-    def tokenize(self, text: str, add_bos: bool = True) -> List[int]:
+    def tokenize(self,
+                 text: Union[str, List[str]],
+                 add_bos: bool = True) -> List[int]:
         '''
         Decode the id to words
 
-        :param text: The text to be tokenized
+        :param text: The text or batch of text to be tokenized
         :param add_bos:
 
         :return: list of ids that indicates the tokens
         '''
-        if isinstance(text, str):
-            bstr = text.encode()
-        else:
-            bstr = text
-        return self._tokenize(bstr, add_bos)
+        is_batched = True if isinstance(text, (list, tuple)) else False
+        if not is_batched:
+            text = [text]
+
+        result = []
+        for t in text:
+            if isinstance(t, str):
+                bstr = t.encode()
+            else:
+                bstr = t
+            result.append(self._tokenize(bstr, add_bos))
 
-    def decode(self, tokens: List[int]) -> str:
+        if not is_batched:
+            result = result[0]
+        return result
+
+    def decode(self,
+               tokens: Union[List[int], List[List[int]], "np.ndarray"]) -> str:
```

Comment:
should we rename this to `batch_decode`?

Reply:
Yes, `batch_decode` is better.